firmware/src/apps/light_dimmer/light_dimmer.cpp

```diff
@@ -269,7 +269,7 @@ void LightDimmerApp::updateStateFromHASS(MQTTStateUpdate mqtt_state_update)
         }
     }
 
-    if (rgb_color != NULL)
+    if (rgb_color != NULL && !cJSON_IsNull(rgb_color))
```

Comment:
please, can we rewrite complex if statemens in a more readible way. in this case, never use ! on a condition of an end.

Something like rgb_color != NULL && cJSON_IsNull(rgb_color) != NULL is a lot more readable.